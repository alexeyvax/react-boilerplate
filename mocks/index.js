
export const data = {
  1: 'one',
  2: 'two',
  3: 'three',
};

export const initStateGetData = {
  data: {},
};
