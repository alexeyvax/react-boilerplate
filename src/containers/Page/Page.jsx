import React from 'react';
import propTypes from 'prop-types';

import Title from '../../components/Title';
import Button from '../../components/Button';

const Page = ({
  data, name, onGetData, onClearData,
}) => {
  const listOfData = Object.keys(data);
  return (
    <React.Fragment>
      {name && <Title name={name} />}
      <Button content="Get data" className="testing" handler={onGetData} />
      {listOfData.length > 0 &&
        <React.Fragment>
          <ol>
            {listOfData.map(key => (
              <li key={key}>{data[key]}</li>
            ))}
          </ol>
          <Button content="Clear data" handler={onClearData} />
        </React.Fragment>
      }
    </React.Fragment>
  );
};

Page.propTypes = {
  data: propTypes.objectOf(propTypes.string).isRequired,
  name: propTypes.string,
  onGetData: propTypes.func.isRequired,
  onClearData: propTypes.func.isRequired,
};

Page.defaultProps = {
  name: null,
};

export default Page;
