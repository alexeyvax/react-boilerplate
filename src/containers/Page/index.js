import { connect } from 'react-redux';
import Page from './Page';
import { getDataSelector } from '../../selectors';
import { getData, clearData } from '../../actions/creators';

const mapStateToProps = state => ({
  data: getDataSelector(state),
  name: state.data.name,
});

const mapDispatchToProps = {
  onGetData: getData,
  onClearData: clearData,
};

export default connect(mapStateToProps, mapDispatchToProps)(Page);
