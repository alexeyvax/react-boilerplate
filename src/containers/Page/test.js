import React from 'react';
import renderer from 'react-test-renderer';

import { data } from '../../../mocks';
import Page from './Page';

it('renders correctly without optional props', () => {
  const props = {
    data,
    onGetData: jest.fn(),
    onClearData: jest.fn(),
  };
  const tree = renderer
    .create(<Page {...props} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with optional props', () => {
  const props = {
    data,
    name: 'test-name',
    onGetData: jest.fn(),
    onClearData: jest.fn(),
  };
  const tree = renderer
    .create(<Page {...props} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with empty data prop', () => {
  const props = {
    data: {},
    onGetData: jest.fn(),
    onClearData: jest.fn(),
  };
  const tree = renderer
    .create(<Page {...props} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
