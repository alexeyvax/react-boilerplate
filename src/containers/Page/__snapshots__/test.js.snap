// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty data prop 1`] = `
<button
  className="testing"
  onClick={[MockFunction]}
  type="button"
>
  Get data
</button>
`;

exports[`renders correctly with optional props 1`] = `
Array [
  <h1
    className="test"
  >
    Hello test-name
  </h1>,
  <button
    className="testing"
    onClick={[MockFunction]}
    type="button"
  >
    Get data
  </button>,
  <ol>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ol>,
  <button
    className="null"
    onClick={[MockFunction]}
    type="button"
  >
    Clear data
  </button>,
]
`;

exports[`renders correctly without optional props 1`] = `
Array [
  <button
    className="testing"
    onClick={[MockFunction]}
    type="button"
  >
    Get data
  </button>,
  <ol>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ol>,
  <button
    className="null"
    onClick={[MockFunction]}
    type="button"
  >
    Clear data
  </button>,
]
`;
